text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,770 @@
+    constitute a high-priority bug in either the emulation or the
+    implementation of the target.
+  - If it is not possible to run the target via emulation, these resources must
+    additionally be sufficient for the Rust infrastructure team to make them
+    available for access by Rust team members, for the purposes of development
+    and testing. (Note that the responsibility for doing target-specific
+    development to keep the target well maintained remains with the target
+    maintainers. This requirement ensures that it is possible for other
+    Rust developers to test the target, but does not obligate other Rust
+    developers to make target-specific fixes.)
+  - Resources provided for CI and similar infrastructure must be available for
+    continuous exclusive use by the Rust project. Resources provided
+    for access by Rust team members for development and testing must be
+    available on an exclusive basis when in use, but need not be available on a
+    continuous basis when not in use.
+- Tier 1 targets must not have a hard requirement for signed, verified, or
+  otherwise "approved" binaries. Developers must be able to build, run, and
+  test binaries for the target on systems they control, or provide such
+  binaries for others to run. (Doing so may require enabling some appropriate
+  "developer mode" on such systems, but must not require the payment of any
+  additional fee or other consideration, or agreement to any onerous legal
+  agreements.)
+  - The Rust project may decide to supply appropriately signed binaries if
+    doing so provides a smoother experience for developers using the target,
+    and a tier 2 target with host tools already requires providing appropriate
+    mechanisms that enable our infrastructure to provide such signed binaries.
+    However, this additional tier 1 requirement ensures that Rust developers
+    can develop and test Rust software for the target (including Rust itself),
+    and that development or testing for the target is not limited.
+- All requirements for tier 2 apply.
+
+A tier 1 target may be demoted if it no longer meets these requirements but
+still meets the requirements for a lower tier. Any proposal for demotion of a
+tier 1 target requires a full RFC process, with approval by the compiler and
+release teams. Any such proposal will be communicated widely to the Rust
+community, both when initially proposed and before being dropped from a stable
+release. A tier 1 target is highly unlikely to be directly removed without
+first being demoted to tier 2 or tier 3. (The amount of time between such
+communication and the next stable release may depend on the nature and severity
+of the failed requirement, the timing of its discovery, whether the target has
+been part of a stable release yet, and whether the demotion or removal can be a
+planned and scheduled action.)
+
+Raising the baseline expectations of a tier 1 target (such as the minimum CPU
```

Comment:
this paragraph doesn't state how much community communication is expected for raising baseline expectations of a tier 1 target.

Given that the previous paragraph makes a point of saying that demotion *requires* a full RFC, I suspect the intent is *not* to imply that the same level of communication written in the previous paragraph is meant to apply here.

I personally am fine with leaving the amount of expected communication unspecified for now.

I personally would like to try to do a better job within the compiler of issuing future-incompatibility warnings from the compiler itself when it is invoked for a host (or host+target combination) atop (or explicitly targeting) a CPU or OS version that is scheduled to be dropped from tier 1 support. But I don't want to block this RFC based upon that.